### PR TITLE
Orafin Payment Models

### LINKED ISSUE #675: create an OraFin payment data model
Using the [mapping file](https://docs.google.com/spreadsheets/d/15vSge029GYl3C9_g_izyZFbGRKt8a5j6BMBu8XETtUI/edit?usp=sharing) and "[Things we need to add to our code spec](https://docs.google.com/document/d/1iIfLXn4yy_KXeECsE-qiu-WTW9yoHe4XfOQUZOwvFBw/edit#bookmark=id.wofmdf5872m5)", create a data model for an OraFin invoice with invoice lines.

### PR DESCRIPTION
**Pull request by jermnelson**
Fixes #675 

Adds models for Orafin using the [attrs](https://www.attrs.org/en/stable/) package that was already part of the libsys_airflow dependencies. Using the accompanying [cattrs](https://github.com/python-attrs/cattrs) package, Okapi JSON payloads for invoices, invoice-lines, purchase order lines, funds, and organizations are used to populate the classes in this PR in the `payments_to_orafin` DAG.

The models will likely need some adjustment for work on generating the actual feeder file as part of ticket [677](https://github.com/sul-dlss/libsys-airflow/issues/677).

### CODE REVIEW

File: libsys_airflow/plugins/folio/helpers/orafin_models.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    def invoice_type(self):
+        if self.amount < 0:
+            return "CR"
+        return "DR"
+
+    @property
+    def terms_name(self):
+        if self.paymentDate:
+            return "IMMEDIATE"
+        return "N30"
```
Comment (shelleydoljack) [inline libsys_airflow/plugins/folio/helpers/orafin_models.py:106]: Need to add another property `attachment_flag` for WILLCALL invoices. Could be in this PR or I can add a ticket to address it in another PR.

Comment (shelleydoljack) [reply libsys_airflow/plugins/folio/helpers/orafin_models.py:106]: The field in the invoice json is called `paymentTerms`.

**Review by shelleydoljack — APPROVED** [review]
Awesome work! One comment about the attachment flag missing for WILLCALL invoices.

File: libsys_airflow/plugins/folio/helpers/orafin_models.py (hunk 1 above)
Comment (jermnelson) [reply libsys_airflow/plugins/folio/helpers/orafin_models.py:106]: Thanks @shelleydoljack; do you have an example of a WILLCALL invoice I could test with adding the `attachment_flag` property?

Comment (shelleydoljack) [reply libsys_airflow/plugins/folio/helpers/orafin_models.py:106]: https://folio-test.stanford.edu/invoice/view/7393322f-aeaa-4d51-84b6-b83d335f9a33

**Pull request merged**